nil/common/concurrent/utils.go

```diff
@@ -71,29 +80,40 @@ func RunWithTimeout(ctx context.Context, timeout time.Duration, fs ...FuncWithSo
 
 	var once sync.Once
 	var originError error
-	for _, f := range fs {
+	for _, t := range tasks {
 		wg.Add(1)
 
-		go func(fn FuncWithSource) {
+		go func(task Task) {
+			taskName := task.TaskName
+			rootContextName, ok := ctx.Value(RootContextNameLabel).(string)
+			if !ok {
+				rootContextName = "<unknown>"
+			}
+			debugger.SetLabels(func() []string {
+				return []string{
+					TaskNameLabel, taskName,
+					RootContextNameLabel, rootContextName,
+				}
+			})
 			defer wg.Done()
-			if err := fn.Func(ctx); err != nil {
+			if err := task.Func(ctx); err != nil {
 				once.Do(func() {
 					originError = &ExecutionError{
 						Err:   err,
-						Stack: fn.Stack,
+						Stack: task.Stack,
 					}
-					cancel()
+					cncl()
```

Comment:
build fails here

Reply:
Thanks, I added to the commit line by line